### PR TITLE
[DCOS_OSS-618] Ignore core.ssl_verify for external requests

### PR DESCRIPTION
**Pull request by bamarni**
Otherwise, when a cluster is configured with a custom CA path, it will
try to use it instead of the system's CA bundle and external requests
will fail.

In such cases, setting it to None will let the requests package enabling
it by default.

### CODE REVIEW

File: dcos/http.py [hunk 1]
```diff
@@ -190,6 +210,7 @@ def _request_match(expected_url, actual_url):
     if is_success(response.status_code):
         return response
     elif response.status_code == 401:
+        prompt_login = config.get_config_val("core.prompt_login", toml_config)
```
Comment (klueska) [inline dcos/http.py]: This change seems unrelated. It should either be a second commit in this pull request with a detailed commit message describing its purpose, or a separate PR.

File: dcos/http.py [hunk 2]
```diff
@@ -27,6 +27,34 @@ def _default_is_success(status_code):
 
     return 200 <= status_code < 300
 
+def _is_request_to_cluster(url, toml_config=None):
```
Comment (klueska) [inline dcos/http.py]: Can we call this `_is_request_to_dcos()` instead?

**Comment by klueska** [comment]
Also, we need to add an integration test before we can call this *done*.

**Comment by bamarni** [comment]
I've just pushed a commit according to your comments.

What kind of integration test are you thinking of? As it involves an HTTPS request to an external URL I don't see an obvious way to do it.

File: dcos/http.py [hunk 3]
```diff
@@ -27,6 +27,34 @@ def _default_is_success(status_code):
...
+    :type toml_config: Toml
+    :return: whether the request is for the cluster
+    :rtype: bool
+    """
+
+    if toml_config is None:
+        toml_config = config.get_config()
+
+    dcos_url = urlparse(config.get_config_val("core.dcos_url", toml_config))
+    cosmos_url = urlparse(
```
Comment (klueska) [inline dcos/http.py]: What is `cosmos_url`? And what happens if it's not set in the toml file (as far as I know, it's usually not set).

File: dcos/http.py [hunk 4]
```diff
@@ -27,6 +27,34 @@ def _default_is_success(status_code):
...
+
+    # request should match scheme + netloc
+    def _request_match(expected_url, actual_url):
+        return expected_url.scheme == actual_url.scheme and \
+                    expected_url.netloc == actual_url.netloc
+
+    request_to_cluster = _request_match(dcos_url, parsed_url) or \
+        _request_match(cosmos_url, parsed_url)
+
+    return request_to_cluster
```
Comment (klueska) [inline dcos/http.py]: You can probably just return the or statement directly without assigning to an intermediate variable? Otherwise, you should rename this variable to something like `is_request_to_dcos`. Up to you.

File: dcos/http.py [hunk 5]
```diff
@@ -86,7 +114,11 @@ def _request(method,
     if 'headers' not in kwargs:
         kwargs['headers'] = {'Accept': 'application/json'}
 
-    verify = _verify_ssl(verify, toml_config)
+    if _is_request_to_dcos(url, toml_config):
+        verify = _verify_ssl(verify, toml_config)
+    else:
+        # Ignore core.ssl_verify if request is outside the DC/OS cluster (DCOS_OSS-618)
+        verify = None
```
Comment (klueska) [inline dcos/http.py]: Instead of doing an if/else here, I would just do:
```
if verify and _is_request_to_dcos(url, toml_config):
    verify = _verify_ssl(verify, toml_config)
```

File: dcos/http.py (hunk 3 above)
Comment (bamarni) [inline dcos/http.py]: I also don't have it set, in that case it is ignored, urlparse will return an empty ParseResult and this won't be matched.

File: dcos/http.py (hunk 4 above)
Comment (bamarni) [inline dcos/http.py]: I'll set it as `is_request_to_dcos` for readability.

File: dcos/http.py (hunk 5 above)
Comment (bamarni) [inline dcos/http.py]: Here `verify` can be a path, and even though I coulnd't see it in the codebase, my concern was if this function gets called with an explicit custom CA path for the cluster and an external https url. In such cases I wanted to force verify to `None`.
